src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+
+    app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                 &trajectory);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path_parameters.path, path_parameters.t_start, path_parameters.t_end,
+        arc_length_param);
+
+    const float arc_segment_length =
+        arc_length_param.arc_length_values[arc_length_param.num_values - 1];
+
+    float segment_length_sum = 0;
+
+    // Test by checking the length between points
+    // and the length of the total path
+    for (uint i = 0; i < trajectory.num_elements - 1; i++)
+    {
+        float length = static_cast<float>(
+            sqrt(pow(trajectory.trajectory_elements[i].position.x -
+                         trajectory.trajectory_elements[i + 1].position.x,
+                     2) +
+                 pow(trajectory.trajectory_elements[i].position.y -
+                         trajectory.trajectory_elements[i + 1].position.y,
+                     2)));
+        segment_length_sum += length;
+    }
+    EXPECT_NEAR(segment_length_sum, arc_segment_length, 0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.y,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).y,
+        0.01);
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[0].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_start)
+            .x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[0].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_start)
+            .y,
+        0.01);
+}
+
+TEST_F(TrajectoryPlannerTest, dynamics_dont_exceed_maximums_straight_line)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {0, 0, 1, 0}},
+        .y = {.coefficients = {0, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 3;
+    path_parameters.num_segments               = 500;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 9;
+    path_parameters.initial_speed              = 2;
```

Comment:
Seems like the `initial`, `final`, and `max_allowable` speed / acceleration values are all important for dynamics tests (both this one and the curved path one). Are there any reasons you chose the values you did for these tests as they are now? Would it be worth making several tests for each case with different parameters that affect dynamics?

Reply:
Made 3 more tests for different initial/final velocities and max accelerations for straight lines paths